src/Core/src/Platform/iOS/MauiLabel.cs

```diff
@@ -68,16 +65,25 @@ public override void LayoutSubviews()
 				case Maui.TextAlignment.Start:
 					fitSize = SizeThatFits(_label.Frame.Size.ToCGSize());
 					labelHeight = (nfloat)Math.Min(Bounds.Height, fitSize.Height);
-					Frame = new RectangleF(labelX, labelY, labelWidth, labelHeight);
+					var startFrame = new RectangleF(labelX, labelY, labelWidth, labelHeight);
+
+					if (startFrame != RectangleF.Empty)
+						Frame = startFrame;
 					break;
 				case Maui.TextAlignment.Center:
-					Frame = new RectangleF(labelX, labelY, labelWidth, labelHeight);
+					var centerFrame = new RectangleF(labelX, labelY, labelWidth, labelHeight);
+
+					if (centerFrame != RectangleF.Empty)
+						Frame = centerFrame;
 					break;
 				case Maui.TextAlignment.End:
 					fitSize = SizeThatFits(_label.Frame.Size.ToCGSize());
-					labelHeight = (nfloat)Math.Min(Bounds.Height, fitSize.Height);
-					nfloat yOffset = (nfloat)(_label.Height - labelHeight);
-					Frame = new RectangleF(labelX, yOffset, labelWidth, labelHeight);
+					var fitHeight = (nfloat)Math.Min(Bounds.Height, fitSize.Height);
+					nfloat yOffset = labelHeight - fitHeight;
+					var endFrame = new RectangleF(labelX, yOffset, labelWidth, fitHeight);
```

Comment:
Can we use the same vars for consistency?

```suggestion
					labelHeight = (nfloat)Math.Min(Bounds.Height, fitSize.Height);
					labelY = labelHeight - fitHeight;
					var endFrame = new RectangleF(labelX, labelY, labelWidth, labelHeight);
```

Reply:
We need a different temp variable in this case, btu we can rename it.